dangerfile.js

```diff
@@ -1,27 +1,28 @@
-// eslint-disable-next-line import/no-extraneous-dependencies, no-unused-vars, object-curly-newline
-import { danger, warn, fail, message } from 'danger';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { danger, warn, fail } from 'danger';
 
-const newChangelog = danger.git.created_files.filter((filePath) => {
-  const srcFilePattern = /^packages\/terra-([a-z-])*\/CHANGELOG.md/i;
-  return srcFilePattern.test(filePath);
-});
+const determinePackages = (changedList, path) => {
+  const path1 = path.split('packages/')[1];
+  const packageName = path1.split('/')[0];
+  return changedList.includes(packageName) ? changedList : changedList.concat([packageName]);
+};
```

Comment:
This function is a little strange. It is adding items to a parameter. It may be cleaner to remove the adding logic and return the package name, the calling reducer can then add the item.

If we want to keep this function altering params it would be slightly cleaner to make the reducer `changedList` a `Set`. This would remove the lengthy return statement. At the very least we should switch `changedList.concat([packageName])` to `changedList.push(packageName)`. Creating and concatenating an additional array isn't necessary.

Reply:
Here is a rough idea of what I meant by removing the reducers. This is just an example, these changes aren't required. 

https://gist.github.com/StephenEsser/5ac86a603d58e0042e40d381c4fcb224

Reply:
I agree your code made it clearer what was happening. Had to fix an issue with checking files that aren't in any package.